### PR TITLE
Add PyMC code examples to math function docstrings

### PR DESCRIPTION
**Pull request by fonnesbeck**
## Description

Adding usage examples to docstrings

Fixes issue at pymc-devs/pymc#5537

## Related Issue

- [ ] Closes #
- [ ] Related to #

## Checklist

- [ ] Checked that [the pre-commit linting/style checks pass](https://docs.pymc.io/en/latest/contributing/python_style.html)
- [ ] Included tests that prove the fix is effective or that the new feature works
- [ ] Added necessary documentation (docstrings and/or example notebooks)
- [ ] If you are a pro: each commit corresponds to a [relevant logical change](https://wiki.openstack.org/wiki/GitCommitMessages#Structural_split_of_changes)

## Type of change

- [ ] New feature / enhancement
- [ ] Bug fix
- [ ] Documentation
- [ ] Maintenance
- [ ] Other (please specify):

### CODE REVIEW

File: pytensor/tensor/math.py [hunk 1]
```diff
@@ -1945,6 +1945,18 @@ def dot(l, r):
     """Return a symbolic dot product.
 
     This is designed to work with both sparse and dense tensors types.
+
+    Example usage with PyMC:
+
+    .. code:: python
+
+        import pymc3 as pm
+        import theano.tensor as tt
```
Comment (maresb) [inline pytensor/tensor/math.py:1954]: Theano and `tt` below

**Comment by fonnesbeck** [comment]
Going to do a few more before merge, if these look good.

File: pytensor/tensor/basic.py [hunk 2]
```diff
@@ -757,7 +757,32 @@ def cast(x, dtype: str | np.dtype) -> TensorVariable:
...
+        Values selected at `False` elements of `cond`.
+
+    Examples
+    --------
+    This example demonstrates how `switch` can be used in PyMC to model a
+    categorical variable.
+
+    .. code:: python
+
+        with pm.Model():
```
Comment (ricardoV94) [inline pytensor/tensor/basic.py:782]: Import pymc, use pm.math.switch? For a fully reproducible snippet and emphasize math module

File: pytensor/tensor/math.py [hunk 3]
```diff
@@ -1945,6 +1945,18 @@ def dot(l, r):
...
+
+    .. code:: python
+
+        import pymc as pm
+        import pytensor.tensor as pt
+
+        with pm.Model() as model:
+            x = pm.Normal('x', mu=0, sd=1, shape=2)
+            y = pm.Normal('y', mu=0, sd=1, shape=2)
+            z = pt.dot(x, y)
```
Comment (ricardoV94) [inline pytensor/tensor/math.py:1959]: Don't show pytensor?

```suggestion
            z = pt.math.dot(x, y)
```

File: pytensor/tensor/special.py [hunk 4]
```diff
@@ -485,6 +485,36 @@ def c_code_cache_version():
...
+    In PyMC, you can use this function to compute a softmax over a vector of
+    probabilities representing the likelihood of each class in a multiclass
+    classification problem. Here is an example::
+
+        import pymc as pm
+        import pytensor.tensor as pt
+
+        with pm.Model() as model:
+            weights = pm.Gamma('weights', 1, 1, shape=3)
+            softmax_prob = pt.softmax(weights)
```
Comment (ricardoV94) [inline pytensor/tensor/special.py:514]: Same

File: pytensor/tensor/basic.py (hunk 2 above)
Comment (fonnesbeck) [reply pytensor/tensor/basic.py:782]: Should we use `pm.math` or `pt`, since this is in the pytensor repo?

File: pytensor/tensor/math.py (hunk 3 above)
Comment (fonnesbeck) [reply pytensor/tensor/math.py:1959]: See comment above.

File: pytensor/tensor/basic.py (hunk 2 above)
Comment (ricardoV94) [reply pytensor/tensor/basic.py:784]: Sorry I didn't even realize this was in the pytensor repo. These examples should be in PyMC and in the PyMC math module docs. It doesn't make sense to mention PyMC in PyTensor

**Comment by ricardoV94** [comment]
Besides the question of adding PyMC-centric examples, we should definitely scratch the default Elemwise docstrings that we are getting now and which you see when you go to the math module in PyMC. That's just useless gibberish.

Mentioned in the understated: https://github.com/pymc-devs/pytensor/issues/292

**Comment by fonnesbeck** [comment]
Might be easiest then to have explicit wrappers to dispatch to PyTensor and override the docstrings.

**Pull request closed without merging**